src/crimson/os/seastore/extent_placement_manager.cc

```diff
@@ -526,6 +526,12 @@ ExtentPlacementManager::write_preallocated_ool_extents(
       auto& extents = p.second;
       assert(writer->get_type() == backend_type_t::RANDOM_BLOCK);
       return writer->alloc_write_ool_extents(t, extents);
+    }).finally([this, &t] {
+      if (t.is_conflicted() && t.is_issued_ool_writes()) {
```

Comment:
Nice catch!

But probably need to find another way, also see https://github.com/ceph/ceph/pull/59367 -- transaction retry won't wait for ool writes, which means at the time, `Transaction::reset_preserve_handle()` may be called and the `Transaction` already reset and reused.

Reply:
What do you think the recent update I posted?